deployment/ccip/changeset/v1_6/cs_ccip_home.go

```diff
@@ -618,12 +609,22 @@ func AddDonAndSetCandidateChangeset(
 	if err != nil {
 		return cldf.ChangesetOutput{}, fmt.Errorf("get node info: %w", err)
 	}
+	nonBootstraps := nodes.NonBootstraps()
 
-	txOpts := e.BlockChains.EVMChains()[cfg.HomeChainSelector].DeployerKey
-	if cfg.MCMS != nil {
-		txOpts = cldf.SimTransactOpts()
+	var expectedDonID uint32
+	if cfg.DonIDOverride != 0 {
+		expectedDonID = cfg.DonIDOverride
+	} else {
+		expectedDonID, err = state.Chains[cfg.HomeChainSelector].CapabilityRegistry.GetNextDONId(&bind.CallOpts{
+			Context: e.GetContext(),
+		})
+		if err != nil {
+			return cldf.ChangesetOutput{}, fmt.Errorf("get next don id: %w", err)
+		}
 	}
-	var donMcmsTxs []mcmstypes.Transaction
+
+	dons := make([]ccipseqs.DONAddition, len(cfg.PluginInfo.OCRConfigPerRemoteChainSelector))
```

Comment:
cfg.PluginInfo.OCRConfigPerRemoteChainSelector
any chance this could be empty?

Reply:
I would say no, otherwise calling this changeset would be a noop